lib/rex/socket.rb

```diff
@@ -809,10 +828,64 @@ def type?
 
 protected
 
-  attr_writer :peerhost, :peerport, :localhost, :localport # :nodoc:
+  attr_writer :peerhost, :peerhostname, :peerport, :localhost, :localport # :nodoc:
   attr_writer :context # :nodoc:
   attr_writer :ipv # :nodoc:
 
+  def self.rex_gethostbyname(name, resolver = @@resolver)
+    raise ::SocketError.new(
+      "Rex::Socket internal DNS resolution requires passing/setting a resolver"
+    ) unless resolver
+    # Pull both record types
+    v4 = resolver.send(name, ::Net::DNS::A).answer.first
+    v6 = resolver.send(name, ::Net::DNS::AAAA).answer.first
+    # Emulate ::Socket's error if no responses found
+    if !v4 and !v6
+      raise ::SocketError.new('getaddrinfo: Name or service not known')
+    end
+    # Build response array
+    hostbyname = [name, []]
+    if v4
+      hostbyname << ::Socket::AF_INET
+      hostbyname << self.addr_aton(v4.address.to_s)
+      hostbyname << self.addr_aton(v6.address.to_s) if v6
+    else
+      hostbyname << ::Socket::AF_INET6
+      hostbyname << self.addr_aton(v6.address.to_s)
+    end
+    return hostbyname
+  end
+
+  def self.rex_getaddrinfo(name, resolver = @@resolver)
```

Comment:
Missing YARD documentation here, which would be quite helpful, particularly for understanding what these parameters types are meant to be.

Reply:
This wasn't marked as resolved but confirmed that the updates fix this issue so marking as resolved.